### PR TITLE
[40] [Backend]  Add Locator service

### PR DESCRIPTION
**Pull request by markgravity**
https://github.com/markgravity/flutter-survey/issues/40

## What happened

We need a service to handle dependency injection

### CODE REVIEW

File: lib/services/locator/locator_service.dart [hunk 1]
```diff
@@ -0,0 +1,3 @@
+import 'package:get_it/get_it.dart';
+
+final locator = GetIt.instance;
```
Comment (sleepylee) [inline lib/services/locator/locator_service.dart:3]: I'm not sure if this is necessary as I saw you already created an instance of it below? 😓

Comment (markgravity) [reply lib/services/locator/locator_service.dart:3]: I just want to create a global variable so I can use like these everywhere
```dart
class ApiServiceImpl implements ApiService {
  static String? _fallbackBaseUrl;
  static String? _fallbackToken;
  final HttpService _httpService = locator.get();
  ...
```

Comment (manh-t) [reply lib/services/locator/locator_service.dart:3]: @markgravity I'm curious too. Can't we use `locator` variable below as a global variable? 🙏 Creating one more variable to hold GetIt instance is just wasting memory 🤔

Comment (markgravity) [reply lib/services/locator/locator_service.dart:3]: @sleepylee , @manh-t my bad. It's redundant code line 😁

Comment (markgravity) [reply lib/services/locator/locator_service.dart:6]: > @markgravity I'm curious too. Can't we use `locator` variable below as a global variable? 🙏 Creating one more variable to hold GetIt instance is just wasting memory 🤔

This `locator` there is a reference to `GetIt.instance`, so I think it won't waste the memory IMO 😀

**Review by manh-t — APPROVED** [review]

**Review by sleepylee — APPROVED** [review]

**Comment by sleepylee** [comment]
tbh we will have to revise the lifecycle of this living instance. Maybe this one is alive throughout the whole app's session, but for the later instances, please recheck if it's necessary to be singleton always 😄  it's a very common pitfall when applying DI

**Pull request merged**